File: rxhttp/src/main/java/com/senierr/http/cookie/CookieStore.java
package com.senierr.http.cookie;

import java.util.List;

import okhttp3.Cookie;
import okhttp3.CookieJar;
import okhttp3.HttpUrl;

/**
 * Cookie仓库接口
 *
 * @author zhouchunjie
 * @date 2018/11/07
 */
public interface CookieStore {

    CookieJar getCookieJar();

    boolean isExpired(Cookie cookie);

    void saveCookies(HttpUrl url, List<Cookie> cookies);

    void saveCookie(HttpUrl url, Cookie cookie);

    List<Cookie> getCookies(HttpUrl url);

    List<Cookie> getAllCookie();

    void removeCookie(HttpUrl url, Cookie cookie);

    void removeCookies(HttpUrl url);

    void clear();
}

File: rxhttp/src/main/java/com/senierr/http/cookie/store/CookieStore.kt
package com.senierr.http.cookie.store

import okhttp3.Cookie
import okhttp3.HttpUrl

/**
 * Cookie仓库接口
 *
 * @author zhouchunjie
 * @date 2018/11/07
 */
interface CookieStore {

    fun isExpired(cookie: Cookie): Boolean

    fun saveCookies(url: HttpUrl, cookies: MutableList<Cookie>)

    fun saveCookie(url: HttpUrl, cookie: Cookie)

    fun getCookies(url: HttpUrl): MutableList<Cookie>

    fun getAllCookie(): MutableList<Cookie>

    fun removeCookie(url: HttpUrl, cookie: Cookie)

    fun removeCookies(url: HttpUrl)

    fun clear()
}